### ISSUE TITLE
grain goes into infinite loop on hello_world

### ISSUE THREAD

**Issue by donpdonp** [grain-lang/grain#1102]
I've been following grain for a while but hadn't installed it a year or so, so I thought I'd download the compiler. The experience was less than ideal and I thought I'd file a ticket on it because first impressions matter.

A new user comes to read https://grain-lang.org/docs/getting_grain. Great lets download it.

```
$ wget -O grain https://github.com/grain-lang/grain/releases/download/grain-v0.4.6/grain-linux-x64
2022-01-13 09:15:28 (19.9 MB/s) - ‘grain’ saved [41746833/41746833]

$ chmod 755 grain
```
okay lets poke at it

```
$ ./grain
error: missing required argument 'file'
```
this is a chance to provide something helpful, like how to get the help screen (--help? -h?)

also lets try to run something that does exist, maybe that'll have more information.

```$ ./grain hello.gr
/snapshot/grain/cli/bin/grainc.js: FILE argument: no `hello.gr' file
Usage: /snapshot/grain/cli/bin/grainc.js [OPTION]... FILE
Try `/snapshot/grain/cli/bin/grainc.js --help' for more information.
child_process.js:682
    throw err;
    ^

Error: Command failed: "/home/donp/desktop/grain" /snapshot/grain/cli/bin/grainc.js --stdlib /snapshot/grain/stdlib hello.gr
    at checkExecSyncError (child_process.js:643:11)
    at Object.execSync (child_process.js:679:15)
    at execSync (pkg/prelude/bootstrap.js:1936:30)
    at Object.execGrainc [as grainc] (/snapshot/grain/cli/bin/exec.js:29:10)
    at module.exports (/snapshot/grain/cli/bin/compile.js:5:8)
    at Command.<anonymous> (/snapshot/grain/cli/bin/grain.js:177:9)
    at Command.listener [as _actionHandler] (/snapshot/grain/node_modules/commander/lib/command.js:473:17)
    at /snapshot/grain/node_modules/commander/lib/command.js:1173:65
    at Command._chainOrCall (/snapshot/grain/node_modules/commander/lib/command.js:1091:12)
    at Command._parseCommand (/snapshot/grain/node_modules/commander/lib/command.js:1173:27) {
  status: 1,
  signal: null,
  output: [ null, null, null ],
  pid: 4726,
  stdout: null,
  stderr: null
}
```
30 lines of internal error messaging to show the one useful line, hello.gr doesn't exist.

but here's why I filed the ticket
```
$ cat > hello.gr
// hello.gr
print("Hello, world!")
$ ./grain hello.gr
[infinite wait]
```

it sits there. forever. This is Ubuntu 21.10 on a Thinkpad T440s.
```
$ ./grain -v
Grain cli 0.4.6
Grain compiler 0.4.5
```

thanks!

**Comment by ospencer** [comment]
Hey @donpdonp, thanks for the issue. I definitely agree that it's less than ideal, and this will be resolved in Grain v0.5.0. It's not actually sitting in an infinite loop, it's just very slow—that binary is a version of the compiler that's been compiled to JS shipped with a Node.js binary. You can build the compiler from source to get a native-speed executable, which is many times faster.

We've got many performance enhancements in the works that'll fix the speed of the JS compiler, and we're working to bring native executables to everyone.

As for the CLI, we'll try to make that experience better 🙂

**Comment by donpdonp** [comment]
Okay I was playing fast and loose with the definition of 'forever' :). Once I gave it some time, it did finish - almost 3min for hello world! I will definitely try to build from source instead. Thanks!

```
$ time ./grain hello.gr
Hello, world!
./grain hello.gr  175.18s user 1.54s system 126% cpu 2:19.54 total
```
